beacon-chain/operations/service.go

```diff
@@ -149,67 +120,32 @@ func (s *Service) AttestationPool(ctx context.Context, requestedSlot uint64) ([]
 		return nil, errors.Wrapf(err, "could not process slots up to %d", requestedSlot)
 	}
 
-	sort.Slice(attestationsFromDB, func(i, j int) bool {
-		return attestationsFromDB[i].Data.Crosslink.Shard < attestationsFromDB[j].Data.Crosslink.Shard
+	sort.Slice(atts, func(i, j int) bool {
+		return atts[i].Data.Target.Epoch < atts[j].Data.Target.Epoch
 	})
 
 	var validAttsCount uint64
-	for _, att := range attestationsFromDB {
-		slot, err := helpers.AttestationDataSlot(bState, att.Data)
-		if err != nil {
-			return nil, errors.Wrap(err, "could not get attestation slot")
-		}
-		// Delete the attestation if the attestation is one epoch older than head state,
-		// we don't want to pass these attestations to RPC for proposer to include.
-		if slot+params.BeaconConfig().SlotsPerEpoch <= bState.Slot {
+	for _, att := range atts {
+		if _, err = blocks.ProcessAttestation(bState, att); err != nil {
```

Comment:
ok can you align this with the beacon propser server ? we also process attestations there too. So now we process them twice whenever we receive them from the attestation pool

Reply:
We shouldn't do this again in proposer server. Will remove it there